.github/workflows/operator-ci.yml

```diff
@@ -50,38 +78,121 @@ jobs:
           driver: docker
           start args: '--addons=ingress'
 
-      - name: Build the Keycloak Docker image
-        run: |
-          cd quarkus
-          cp dist/target/keycloak-*.tar.gz container/
-          cd container
-          eval $(minikube -p minikube docker-env)
-          docker build --build-arg KEYCLOAK_DIST=$(ls keycloak-*.tar.gz) . -t keycloak:${GITHUB_SHA}
-
-      - name: Build a custom pre-augmented Keycloak Docker image
-        working-directory: operator
-        run: |
-          eval $(minikube -p minikube docker-env)
-          ./scripts/build-testing-docker-images.sh ${GITHUB_SHA} keycloak
-
       - name: Test operator running locally
         working-directory: operator
         run: |
           mvn clean verify \
               -Dquarkus.kubernetes.deployment-target=kubernetes \
-              -Doperator.keycloak.image=keycloak:${GITHUB_SHA} \
-              -Doperator.keycloak.image-pull-policy=Never \
-              -Dtest.operator.kubernetes.ip=$(minikube ip) \
-              -Dtest.operator.custom.image=custom-keycloak:${GITHUB_SHA}
+              -Doperator.keycloak.image=${{ secrets.TEST_DOCKER_REGISTRY }}/${{ secrets.TEST_DOCKER_REPOSITORY }}/keycloak:${{ steps.vars.outputs.version }} \
+              -Dtest.operator.custom.image=${{ secrets.TEST_DOCKER_REGISTRY }}/${{ secrets.TEST_DOCKER_REPOSITORY }}/custom-keycloak:${{ steps.vars.outputs.version }} \
+              -Dtest.operator.kubernetes.ip=$(minikube ip)
+
+  test-remote:
+    name: Test remote
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set outputs
+        id: vars
+        run: echo "::set-output name=version::0.0.1-$(git rev-parse --short HEAD)"
+      - name: Update maven settings
+        run: mkdir -p ~/.m2 ; cp .github/settings.xml ~/.m2/
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.JDK_VERSION }}
+          cache: 'maven'
+      - name: Setup Minikube-Kubernetes
+        uses: manusa/actions-setup-minikube@v2.4.3
+        with:
+          minikube version: v1.24.0
+          kubernetes version: v1.22.3
```

Comment:
Just a nitpick, we can handle this separately. We might want to move the versions to some env var as we install minikube three times now (similarly to `JDK_VERSION`).

Reply:
that's a good idea, we should refactor a little more next time we work on this file!